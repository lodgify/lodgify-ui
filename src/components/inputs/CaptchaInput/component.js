import React from 'react';
import PropTypes from 'prop-types';
import { Form, Image } from 'semantic-ui-react';

import { TextInput } from 'inputs/TextInput';

/**
 * A captcha input helps to ensure a user is a real human.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({
  value,
  error,
  image,
  isValid,
  label,
  name,
  onBlur,
  onChange,
}) => (
  <Form.Group widths="equal">
    <Form.Field>
      <Image src={image} />
    </Form.Field>
    <Form.Field>
      <TextInput
        autoComplete="off"
        error={error}
        isValid={isValid}
        label={label}
        name={name}
        onBlur={onBlur}
        onChange={onChange}
        value={value}
      />
    </Form.Field>
  </Form.Group>
);

Component.displayName = 'CaptchaInput';

Component.defaultProps = {
  error: false,
  isValid: false,
  label: '',
  name: '',
  onBlur: Function.prototype,
  onChange: Function.prototype,
  value: undefined,
};

Component.propTypes = {
  /** Is the captcha input in an error state. */
  error: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  /** The source url for the image to display. */
  image: PropTypes.string.isRequired,
  /** Is the captcha input in a valid state. */
  isValid: PropTypes.bool,
  /** The visible label for the text input. */
  label: PropTypes.string,
  /** The name for the captcha input. */
  name: PropTypes.string,
  /**
   * Used internally by `Form` so ignored in the styleguide.
   * @ignore
   */
  onBlur: PropTypes.func,
  /**
   * A function called when the captcha input value changes
   * @param {String} name
   * @param {String} value
   */
  onChange: PropTypes.func,
  /** The current value of the input where it is consumed as a controlled component. */
  value: PropTypes.oneOfType([
    PropTypes.bool,
    PropTypes.string,
    PropTypes.number,
  ]),
};
