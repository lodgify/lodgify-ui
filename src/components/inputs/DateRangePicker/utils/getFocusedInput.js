/**
 * @typedef {Object} null
 * @param   {string|null|undefined} controlledFocusedInput
 * @param   {string|null} uncontrolledFocusedInput
 * @return  {string|null}
 */
export const getFocusedInput = (
  controlledFocusedInput,
  uncontrolledFocusedInput
) =>
  typeof controlledFocusedInput === 'undefined'
    ? uncontrolledFocusedInput
    : controlledFocusedInput;
