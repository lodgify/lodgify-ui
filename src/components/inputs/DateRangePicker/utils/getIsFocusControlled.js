/**
 * @typedef {Object} null
 * @param   {string|null|undefined} propsFocusedInput
 * @return  {boolean}
 */
export const getIsFocusControlled = propsFocusedInput =>
  typeof propsFocusedInput !== 'undefined';
