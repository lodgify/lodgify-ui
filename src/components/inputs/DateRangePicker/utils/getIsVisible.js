/**
 * @param  {Object} element
 * @return {boolean}
 */
export const getIsVisible = element => !!element && !!element.offSetParent;
