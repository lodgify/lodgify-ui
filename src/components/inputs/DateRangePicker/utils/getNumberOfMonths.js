import { MAXIMUM_SCREEN_WIDTH_FOR_TWO_MONTH_CALENDAR } from '../constants';

/**
 * @param  {number} windowInnerWidth
 * @return {number|undefined}
 */
export const getNumberOfMonths = windowInnerWidth =>
  windowInnerWidth < MAXIMUM_SCREEN_WIDTH_FOR_TWO_MONTH_CALENDAR
    ? 1
    : undefined;
