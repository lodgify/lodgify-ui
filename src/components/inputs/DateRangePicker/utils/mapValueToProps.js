/**
 * @param  {Object} value
 * @return {Object}
 */
export const mapValueToProps = value =>
  !!value
    ? { endDate: value.endDate, startDate: value.startDate }
    : { startDate: null, endDate: null };
