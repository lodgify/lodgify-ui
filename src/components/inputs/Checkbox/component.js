import React from 'react';
import PropTypes from 'prop-types';
import { Checkbox } from 'semantic-ui-react';

import { InputController } from '../InputController';

import { mapValueToProps } from './utils/mapValueToProps';
import { adaptOnChangeEvent } from './utils/adaptOnChangeEvent';

/**
 * A checkbox allows a user to select a value from a small set of options, often binary.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({
  error,
  isDisabled,
  isChecked,
  isRadioButton,
  isToggle,
  isValid,
  label,
  name,
  onChange,
}) => (
  <InputController
    adaptOnChangeEvent={adaptOnChangeEvent}
    error={error}
    isValid={isValid}
    label={label}
    mapValueToProps={mapValueToProps}
    name={name}
    onChange={onChange}
    value={isChecked}
  >
    <Checkbox disabled={isDisabled} radio={isRadioButton} toggle={isToggle} />
  </InputController>
);

Component.displayName = 'Checkbox';

Component.defaultProps = {
  error: false,
  isChecked: undefined,
  isDisabled: false,
  isToggle: false,
  isRadioButton: false,
  isValid: false,
  label: '',
  name: '',
  onChange: Function.prototype,
};

Component.propTypes = {
  /** Is the checkbox input in an error state. */
  error: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  /** Is the checkbox checked. */
  isChecked: PropTypes.bool,
  /** Is the checkbox disabled. */
  isDisabled: PropTypes.bool,
  /** Is the checkbox formatted as a radio button. Prefer using the <RadioButton> component instead */
  isRadioButton: PropTypes.bool,
  /** Format to show an on or off choice. */
  isToggle: PropTypes.bool,
  /** Is the checkbox in a valid state. */
  isValid: PropTypes.bool,
  /** The label for the checkbox */
  label: PropTypes.node,
  /** The HTML input name. */
  name: PropTypes.string,
  /**
   * Event called when the checkbox value changes.
   * @param {String} name
   * @param{boolean} value
   */
  onChange: PropTypes.func,
};
