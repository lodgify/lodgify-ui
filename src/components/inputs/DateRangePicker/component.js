import moment from 'moment';
import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { uniqueId, debounce } from 'lodash';
import { DateRangePicker } from 'react-dates';

import 'react-dates/initialize';

import { getUpOrDownFromBoolean } from 'utils/get-up-or-down-from-boolean';
import { withResponsive } from 'utils/with-responsive';
import { Icon, ICON_NAMES } from 'elements/Icon';
import { InputController } from 'inputs/InputController';
import { isBlurEvent } from 'utils/is-blur-event';
import { returnFirstArgument } from 'utils/return-first-argument';
import { getWindowHeight } from 'utils/get-window-height';
import { isDisplayedAsModal } from 'utils/is-displayed-as-modal';

import { mapValueToProps } from './utils/mapValueToProps';
import { getNumberOfMonths } from './utils/getNumberOfMonths';
import { MAXIMUM_SCREEN_WIDTH_FOR_TWO_MONTH_CALENDAR } from './constants';

/**
 * A date range picker lets a user pick a date range.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
class Component extends PureComponent {
  state = {
    focusedInput: null,
    windowHeight: null,
  };

  componentDidMount = () => {
    moment.locale(this.props.localeCode);
    this.handleHeightChange();
    global.addEventListener('resize', debounce(this.handleHeightChange, 150));
  };

  componentDidUpdate = (prevProps, prevState) => {
    const { focusedInput: prevFocusedInput } = prevState;
    const { focusedInput } = this.state;
    const { onBlur } = this.props;

    isBlurEvent(prevFocusedInput, focusedInput) && onBlur();
  };

  handleFocusChange = focusedInput => {
    this.setState({ focusedInput });
  };

  handleHeightChange = () => {
    const windowHeight = getWindowHeight();

    if (windowHeight !== this.state.windowHeight) {
      this.setState({
        windowHeight,
      });
    }
  };

  render() {
    const {
      displayFormat,
      endDatePlaceholderText,
      error,
      getIsDayBlocked,
      isValid,
      name,
      onChange,
      startDatePlaceholderText,
      value,
      willOpenAbove,
      windowInnerWidth,
    } = this.props;
    const { focusedInput } = this.state;

    return (
      <InputController
        adaptOnChangeEvent={returnFirstArgument}
        error={error}
        initialValue={null}
        inputOnChangeFunctionName="onDatesChange"
        isFocused={!!focusedInput}
        isValid={isValid}
        mapValueToProps={mapValueToProps}
        name={name}
        onChange={onChange}
        value={value}
      >
        <DateRangePicker
          /* eslint-disable react/jsx-sort-props */
          // Consumer defined props.
          displayFormat={displayFormat}
          endDatePlaceholderText={endDatePlaceholderText}
          isDayBlocked={getIsDayBlocked}
          openDirection={getUpOrDownFromBoolean(willOpenAbove)}
          startDatePlaceholderText={startDatePlaceholderText}
          // Controlled props
          focusedInput={focusedInput}
          // NOTE onDatesChange is required by DateRangePicker but is set in `InputController`
          onDatesChange={Function.prototype}
          onFocusChange={this.handleFocusChange}
          // Static required props.
          endDateId={uniqueId('end_date_id_')}
          startDateId={uniqueId('start_date_id_')}
          // Static custom appearance props.
          customArrowIcon={<Icon name={ICON_NAMES.ARROW_RIGHT} />}
          customInputIcon={<Icon name={ICON_NAMES.CALENDAR} />}
          daySize={52}
          hideKeyboardShortcutsPanel
          navNext={<Icon name={ICON_NAMES.ARROW_RIGHT} />}
          navPrev={<Icon name={ICON_NAMES.ARROW_LEFT} />}
          numberOfMonths={getNumberOfMonths(windowInnerWidth)}
          withPortal={isDisplayedAsModal(this.state.windowHeight)}
          /* eslint-enable react/jsx-sort-props */
        />
      </InputController>
    );
  }
}

Component.displayName = 'DateRangePicker';

Component.defaultProps = {
  displayFormat: 'DD/MM/YYYY',
  endDatePlaceholderText: '',
  error: false,
  getIsDayBlocked: Function.prototype,
  isValid: false,
  localeCode: 'en',
  name: '',
  onBlur: Function.prototype,
  onChange: Function.prototype,
  startDatePlaceholderText: '',
  value: undefined,
  willOpenAbove: false,
  windowInnerWidth: MAXIMUM_SCREEN_WIDTH_FOR_TWO_MONTH_CALENDAR,
};

Component.propTypes = {
  /** The format for inputting and displaying dates as text. */
  displayFormat: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
  /** The visible placeholder text for the end date input. */
  endDatePlaceholderText: PropTypes.string,
  /** Is the date range picker in an error state. A string is displayed as an error message. */
  error: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  /**
   * A function called for each day to be displayed. Returning true blocks that day in the date range picker.
   * @param   {Moment}  day - The day to test.
   * @returns {boolean}     - Is the day blocked.
   */
  getIsDayBlocked: PropTypes.func,
  /** Is the date range picker in a valid state. */
  isValid: PropTypes.bool,
  /** The ISO 639-1 locale code which changes the format and language of days of the week and the months of the year. */
  localeCode: PropTypes.string,
  /** The name for the date range picker. */
  name: PropTypes.string,
  /**
   * Used internally by `Form` so ignored in the styleguide.
   * @ignore
   */
  onBlur: PropTypes.func,
  /**
   * A function called when the date range picker value changes. Dates are wrapped as [Moment objects](https://momentjs.com/docs/#/parsing/).
   * @param {String} name   - `this.props.name`
   * @param {Object} value
   * @param {Moment} value.endDate
   * @param {Moment} value.startDate
   */
  onChange: PropTypes.func,
  /** The visible placeholder text for the start date input. */
  startDatePlaceholderText: PropTypes.string,
  /** The current value of the input where it is consumed as a controlled component. */
  value: PropTypes.shape({
    endDate: PropTypes.object,
    startDate: PropTypes.object,
  }),
  /** Will the calendar open above the input. */
  willOpenAbove: PropTypes.bool,
  /**
   * Is the user on a mobile device.
   * Provided by `withResponsive` so ignored in the styleguide.
   * @ignore
   */
  windowInnerWidth: PropTypes.number,
};

export const ComponentWithResponsive = withResponsive(Component);
