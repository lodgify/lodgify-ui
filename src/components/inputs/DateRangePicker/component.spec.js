jest.mock('utils/get-window-height');
jest.mock('lodash/debounce');

import React from 'react';
import { shallow } from 'enzyme';
import { DateRangePicker as ReactDatesDateRangePicker } from 'react-dates';
import { Responsive } from 'semantic-ui-react';
import moment from 'moment';
import {
  expectComponentToBe,
  expectComponentToHaveChildren,
  expectComponentToHaveProps,
  expectComponentToHaveDisplayName,
} from '@lodgify/enzyme-jest-expect-helpers';
import { debounce } from 'lodash';

import { InputController } from 'inputs/InputController';
import { Icon, ICON_NAMES } from 'elements/Icon';
import { returnFirstArgument } from 'utils/return-first-argument';
import { getWindowHeight } from 'utils/get-window-height';

import { ComponentWithResponsive as DateRangePicker } from './component';

const STARTING_WINDOW_HEIGHT = 900;
const NEXT_WINDOW_HEIGHT = 800;

const getDateRangePicker = () => shallow(<DateRangePicker />);
const getWrappedDateRangePicker = props => {
  const Child = getDateRangePicker().prop('as');

  return shallow(<Child {...props} />);
};

describe('<DateRangePicker />', () => {
  beforeEach(() => {
    getWindowHeight.mockReset();
    getWindowHeight.mockReturnValueOnce(STARTING_WINDOW_HEIGHT);
    getWindowHeight.mockReturnValueOnce(NEXT_WINDOW_HEIGHT);
  });

  it('should be wrapped in a Semantic UI `Responsive` component', () => {
    const wrapper = getDateRangePicker();

    expectComponentToBe(wrapper, Responsive);
  });

  describe('on mount', () => {
    it('should set the correct `moment.locale`', () => {
      const localeCode = 'ko';
      const dateRangePicker = getWrappedDateRangePicker({ localeCode });

      dateRangePicker.instance().componentDidMount();
      const actual = moment.locale();

      expect(actual).toEqual(localeCode);
    });

    it('should call `this.handleHeightChange`', () => {
      const wrapper = getWrappedDateRangePicker();

      wrapper.instance().handleHeightChange = jest.fn();
      wrapper.update();
      wrapper.instance().componentDidMount();

      expect(wrapper.instance().handleHeightChange).toHaveBeenCalled();
    });

    it('should bind handleHeightChange to global resize event', () => {
      global.addEventListener = jest.fn();
      const wrapper = getWrappedDateRangePicker();

      debounce.mockReturnValueOnce(wrapper.instance().handleHeightChange, 150);

      wrapper.instance().componentDidMount();
      expect(global.addEventListener).toHaveBeenCalledWith(
        'resize',
        debounce()
      );
    });
  });

  describe('the `addEventListener` method', () => {
    it('should call `debounce` with the right arguments', () => {
      const wrapper = getWrappedDateRangePicker();
      const handler = wrapper.instance().handleHeightChange;

      expect(debounce).toHaveBeenCalledWith(handler, 150);
    });
  });

  describe('the wrapped `DateRangePicker` component', () => {
    it('should be a Semantic UI `InputController`', () => {
      const wrapper = getWrappedDateRangePicker();

      expectComponentToBe(wrapper, InputController);
    });
  });

  describe('the `InputController` component', () => {
    const getInputController = () =>
      getWrappedDateRangePicker().find(InputController);

    it('should get the right props', () => {
      const wrapper = getInputController();

      expectComponentToHaveProps(wrapper, {
        adaptOnChangeEvent: returnFirstArgument,
        error: false,
        inputOnChangeFunctionName: 'onDatesChange',
        isFocused: false,
        isValid: false,
        name: '',
        onChange: expect.any(Function),
      });
    });

    it('should get the right children', () => {
      const wrapper = getInputController();

      expectComponentToHaveChildren(wrapper, ReactDatesDateRangePicker);
    });
  });

  describe('the `react-dates` `DateRangePicker`', () => {
    const getReactDatesDateRangePicker = () =>
      getWrappedDateRangePicker().find(ReactDatesDateRangePicker);

    it('should get the right consumer defined props', () => {
      const wrapper = getReactDatesDateRangePicker();

      expectComponentToHaveProps(wrapper, {
        displayFormat: 'DD/MM/YYYY',
        endDatePlaceholderText: '',
        isDayBlocked: Function.prototype,
        openDirection: expect.any(String),
        startDatePlaceholderText: '',
      });
    });

    it('should get the right controlled props', () => {
      const wrapper = getReactDatesDateRangePicker();

      expectComponentToHaveProps(wrapper, {
        endDate: null,
        focusedInput: null,
        onDatesChange: Function.prototype,
        onFocusChange: expect.any(Function),
        startDate: null,
      });
    });

    it('should get the right static required props', () => {
      const wrapper = getReactDatesDateRangePicker();

      expectComponentToHaveProps(wrapper, {
        endDateId: expect.stringContaining('end_date_id_'),
        startDateId: expect.stringContaining('start_date_id_'),
      });
    });

    it('should get the right static custom appearance props', () => {
      const wrapper = getReactDatesDateRangePicker();

      expectComponentToHaveProps(wrapper, {
        customArrowIcon: <Icon name={ICON_NAMES.ARROW_RIGHT} />,
        customInputIcon: <Icon name={ICON_NAMES.CALENDAR} />,
        daySize: 52,
        hideKeyboardShortcutsPanel: true,
        navNext: <Icon name={ICON_NAMES.ARROW_RIGHT} />,
        navPrev: <Icon name={ICON_NAMES.ARROW_LEFT} />,
        numberOfMonths: ReactDatesDateRangePicker.defaultProps.numberOfMonths,
        withPortal: expect.any(Boolean),
      });
    });
  });

  describe('Interaction: onChange', () => {
    it('should persist the value in component state', () => {
      const now = moment();
      const value = { startDate: now };
      const dateRangePicker = getWrappedDateRangePicker();

      dateRangePicker.instance().handleInputControllerChange(undefined, value);
      const actual = dateRangePicker.state();

      expect(actual).toEqual(expect.objectContaining(value));
    });
  });

  describe('Interaction: onFocusChange', () => {
    it('should persist the value in component state', () => {
      const value = 'startDate';
      const dateRangePicker = getWrappedDateRangePicker();

      dateRangePicker.instance().handleFocusChange(value);
      const actual = dateRangePicker.state();

      expect(actual).toEqual(
        expect.objectContaining({
          focusedInput: value,
        })
      );
    });
  });

  describe('Interaction: onHeightChange', () => {
    describe('if window height has changed', () => {
      it('should persist the value in component state', () => {
        const dateRangePicker = getWrappedDateRangePicker();

        dateRangePicker.instance().handleHeightChange();
        const actual = dateRangePicker.state();

        expect(actual).toEqual(
          expect.objectContaining({
            windowHeight: NEXT_WINDOW_HEIGHT,
          })
        );
      });
    });

    describe('if window height has not changed', () => {
      it('should do nothing', () => {
        getWindowHeight.mockReset();
        getWindowHeight.mockReturnValue(STARTING_WINDOW_HEIGHT);

        const dateRangePicker = getWrappedDateRangePicker();

        dateRangePicker.instance().handleHeightChange();
        const actual = dateRangePicker.state();

        expect(actual).toEqual(
          expect.objectContaining({
            windowHeight: STARTING_WINDOW_HEIGHT,
          })
        );
      });
    });
  });

  describe('State change: value', () => {
    it('should call the function passed as `props.onChange`', () => {
      const handleChange = jest.fn();
      const props = { name: 'winnie', onChange: handleChange };
      const newState = { endDate: null, startDate: moment() };
      const dateRangePicker = getWrappedDateRangePicker(props);

      dateRangePicker.setState(newState);
      expect(handleChange).toHaveBeenCalledWith(
        props.name,
        expect.objectContaining(newState)
      );
    });
  });

  describe('State change: focusedInput', () => {
    describe('if there is a blur event', () => {
      it('should call `props.onBlur`', () => {
        const onBlur = jest.fn();
        const dateRangePicker = getWrappedDateRangePicker({ onBlur });

        const prevState = { focusedInput: 'startDate' };
        const state = { focusedInput: null };

        dateRangePicker.setState(prevState);
        dateRangePicker.setState(state);

        expect(onBlur).toHaveBeenCalled();
      });
    });

    describe('if there is a no blur event', () => {
      it('should note call `props.onBlur`', () => {
        const onBlur = jest.fn();
        const dateRangePicker = getWrappedDateRangePicker({ onBlur });

        const prevState = { focusedInput: null };
        const state = { focusedInput: null };

        dateRangePicker.setState(prevState);
        dateRangePicker.setState(state);

        expect(onBlur).not.toHaveBeenCalled();
      });
    });
  });

  it('should have displayName `DateRangePicker`', () => {
    const component = getDateRangePicker().prop('as');

    expectComponentToHaveDisplayName(component, 'DateRangePicker');
  });
});
