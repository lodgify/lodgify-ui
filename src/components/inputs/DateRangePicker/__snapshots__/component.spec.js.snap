// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateRangePicker /> should render the right structure 1`] = `
<WithResponsive(DateRangePicker)>
  <Responsive
    as={[Function]}
    isUserOnMobile={false}
    onUpdate={[Function]}
    windowInnerWidth={1024}
  >
    <DateRangePicker
      displayFormat="DD/MM/YYYY"
      endDatePlaceholderText=""
      error={false}
      getIsDayBlocked={[Function]}
      isUserOnMobile={false}
      isValid={false}
      localeCode="en"
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      startDatePlaceholderText=""
      willOpenAbove={false}
      windowInnerWidth={1024}
    >
      <InputController
        adaptOnChangeEvent={[Function]}
        error={false}
        icon={null}
        inputOnChangeFunctionName="onDatesChange"
        isFocused={false}
        isValid={false}
        label={null}
        mapValueToProps={[Function]}
        name=""
        onChange={[Function]}
      >
        <Input
          className=""
          iconPosition={null}
          type="text"
        >
          <div
            className="ui input"
          >
            <DateRangePicker
              customArrowIcon={
                <Icon
                  color={null}
                  hasBorder={false}
                  isButton={false}
                  isCircular={false}
                  isColorInverted={false}
                  isDisabled={false}
                  isLabelLeft={false}
                  labelText={null}
                  labelWeight={null}
                  name="arrow right"
                  path={null}
                  size={null}
                />
              }
              customInputIcon={
                <Icon
                  color={null}
                  hasBorder={false}
                  isButton={false}
                  isCircular={false}
                  isColorInverted={false}
                  isDisabled={false}
                  isLabelLeft={false}
                  labelText={null}
                  labelWeight={null}
                  name="calendar"
                  path={null}
                  size={null}
                />
              }
              daySize={52}
              displayFormat="DD/MM/YYYY"
              endDate={null}
              endDateId="end_date_id_3"
              endDatePlaceholderText=""
              focusedInput={null}
              hideKeyboardShortcutsPanel={true}
              isDayBlocked={[Function]}
              key=".2"
              navNext={
                <Icon
                  color={null}
                  hasBorder={false}
                  isButton={false}
                  isCircular={false}
                  isColorInverted={false}
                  isDisabled={false}
                  isLabelLeft={false}
                  labelText={null}
                  labelWeight={null}
                  name="arrow right"
                  path={null}
                  size={null}
                />
              }
              navPrev={
                <Icon
                  color={null}
                  hasBorder={false}
                  isButton={false}
                  isCircular={false}
                  isColorInverted={false}
                  isDisabled={false}
                  isLabelLeft={false}
                  labelText={null}
                  labelWeight={null}
                  name="arrow left"
                  path={null}
                  size={null}
                />
              }
              onDatesChange={[Function]}
              onFocusChange={[Function]}
              openDirection="down"
              startDate={null}
              startDateId="start_date_id_4"
              startDatePlaceholderText=""
              withPortal={false}
            >
              <div />
            </DateRangePicker>
          </div>
        </Input>
      </InputController>
    </DateRangePicker>
  </Responsive>
</WithResponsive(DateRangePicker)>
`;
